Makefile

```diff
@@ -46,7 +46,11 @@ upgrade: ## update the requirements/*.txt files with the latest packages satisfy
 	mv requirements/test.tmp requirements/test.txt
 
 quality: ## check coding style with pycodestyle and pylint
-	tox -e quality
+	pylint section_to_course manage.py setup.py
+	pycodestyle section_to_course tests manage.py setup.py
+	pydocstyle section_to_course tests manage.py setup.py
+	isort --check-only --diff tests test_utils section_to_course manage.py setup.py test_settings.py
+	@echo "Passed!"
```

Comment:
@Agrendalath We're removing this from tox because after about three hours of fighting with it I couldn't get tox to initialize its environment correctly for the purposes of running tests after adding the latest imports from edx-platform.

Reply:
@Kelketek, it's OK to skip edx-platform imports in quality tests. We'll use integration tests to test functionalities, so it's not a big deal. That's what every third-party edx-platform dependency is doing. I'd like to suggest an alternative approach. I tested it locally, and it works correctly.
1. Restore the `tox` content from before this PR.
2. Remove `PYTHONPATH` from there.
3. Remove `changedir` from there (it is a default behavior).
4. Add `# pylint: disable=import-error, import-outside-toplevel` at the top of the `compat.py` file.
5. Use `edx-platform` imports within functions there. I.e. let's say we want to get the course. To do this, we define the following function and import it in other Python files:
   ```python
    def get_course(course_key: CourseLocator):
        """
        Get a course from the modulestore.
        """
        from xmodule.modulestore.django import modulestore
        return modulestore().get_course(course_key)
   ```
6. Add `organizations` to `INSTALLED_APPS` in `test_settings.py`.
7. See my other comments about the dependencies missing in `requirements/base.in`, and run `make ugprade`.
8. Now, running `tox -e quality` outside of the devstack should work as expected.

The same approach should apply to all `edx-platform` imports. This way, we will have all of them in one place, and it will be much easier to maintain their compatibility across multiple Open edX releases.

Reply:
@Agrendalath I've got this working for everything except the tests. For those, we rely a lot on subclassing. At the moment, our unit tests are really integration tests already and don't work without the platform. For those cases I've squelched the pylint import warnings.

You mentioned that other third-party dependencies are using integration tests-- do you have a model for how these are best done for apps like these? If I'd known of one, I'd have done that from the beginning. I think I can convert the existing tests to unit tests without dependency on the platform by mocking things out now that we have compat.py, but I'd definitely like to have integration tests.

Reply:
> I've got this working for everything except the tests. For those, we rely a lot on subclassing. At the moment, our unit tests are really integration tests already and don't work without the platform. For those cases I've squelched the pylint import warnings.

Makes sense.

> If I'd known of one, I'd have done that from the beginning.

Yeah, I should have mentioned this in my previous review. I suppose I thought we won't have so many non-API platform imports.

> You mentioned that other third-party dependencies are using integration tests-- do you have a model for how these are best done for apps like these?

Sorry, that wasn't clear - I was referring to skipping edx-platform imports in quality tests. Some XBlocks are doing integration tests, but they are relying on [the workbench](https://github.com/openedx/xblock-sdk). Most third-party packages that import from the edx-platform [allows imports to fail without raising](https://github.com/open-craft/openedx-completion-aggregator/blob/f09b732dc2eace873c33f89ac286202c6c47c714/completion_aggregator/management/commands/run_aggregator_test.py#L23-L29) (sometimes [they log some warnings](https://github.com/open-craft/openedx-completion-aggregator/blob/a71ab4f077acee57a2b9f60ed1897891ecc5bb92/completion_aggregator/signals.py#L27-L32)) and just mock these calls in the tests.

[edx-sga](https://github.com/mitodl/edx-sga) used [integration tests](https://github.com/mitodl/edx-sga/blob/master/run_devstack_integration_tests.sh) in the past, but these don't seem to be maintained anymore.